ffi/src/lib.rs

```diff
@@ -878,52 +882,50 @@ fn valkey_value_to_command_response(value: Value) -> RedisResult<CommandResponse
             Ok(command_response)
         }
         Value::Array(array) => {
-            let vec: Vec<CommandResponse> = array
+            let vec: Result<Vec<CommandResponse>, RedisError> = array
                 .into_iter()
-                .map(|v| {
-                    valkey_value_to_command_response(v)
-                        .expect("Value couldn't be converted to CommandResponse")
-                })
+                .map(|v| valkey_value_to_command_response(v))
                 .collect();
-            let (vec_ptr, len) = convert_vec_to_pointer(vec);
+            let (vec_ptr, len) = convert_vec_to_pointer(vec?);
             command_response.array_value = vec_ptr;
             command_response.array_value_len = len;
             command_response.response_type = ResponseType::Array;
             Ok(command_response)
         }
         Value::Map(map) => {
-            let result: Vec<CommandResponse> = map
+            let result: Result<Vec<CommandResponse>, RedisError> = map
                 .into_iter()
                 .map(|(key, val)| {
                     let mut map_response = CommandResponse::default();
 
-                    let map_key = valkey_value_to_command_response(key)
-                        .expect("Value couldn't be converted to CommandResponse");
+                    let map_key = match valkey_value_to_command_response(key) {
+                        Ok(map_key) => map_key,
+                        Err(err) => return Err(err),
+                    };
                     map_response.map_key = Box::into_raw(Box::new(map_key));
 
-                    let map_val = valkey_value_to_command_response(val)
-                        .expect("Value couldn't be converted to CommandResponse");
+                    let map_val = match valkey_value_to_command_response(val) {
+                        Ok(map_val) => map_val,
+                        Err(err) => return Err(err),
+                    };
                     map_response.map_value = Box::into_raw(Box::new(map_val));
 
-                    map_response
+                    Ok(map_response)
                 })
-                .collect::<Vec<_>>();
+                .collect::<Result<Vec<CommandResponse>, RedisError>>();
 
-            let (vec_ptr, len) = convert_vec_to_pointer(result);
+            let (vec_ptr, len) = convert_vec_to_pointer(result?);
```

Comment:
This only returns the first error in the collection, but without these changes, we would've only panicked on the first error anyway. This can be changed, if required, but worth discussing first.